### PR TITLE
tests: add test for enabling systemd instance unit

### PR DESCRIPTION
**Pull request by dustymabe**

### CODE REVIEW

**Comment by dustymabe** [comment]
I decided to write this test because I noticed the enablement bits just landed in RHEL for it to all work. Is there a way I can force this test to run on RHCOS to (in downstream CI)?

File: tests/kola/ignition/systemd-enable-instance-unit/test.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+fatal() {
+    echo "$@" >&2
+    exit 1
+}
+
+# make sure the presets worked and the instantiated unit is enabled
+if [ $(systemctl is-enabled echo@foo.service) != 'enabled' ]; then
+    fatal "echo@foo.service systemd unit should be enabled"
+fi
+ok "echo@foo.service systemd unit is enabled"
```
Comment (jlebon) [inline tests/kola/ignition/systemd-enable-instance-unit/test.sh:25]: Shouldn't we instead/also check that the service ran? E.g. by querying the journal.

**Review by jlebon — COMMENTED** [review]
> I decided to write this test because I noticed the enablement bits just landed in RHEL for it to all work. Is there a way I can force this test to run on RHCOS to (in downstream CI)?

Not currently, but you can symlink them in openshift/os. See e.g. https://github.com/openshift/os/pull/510.

File: tests/kola/ignition/systemd-enable-instance-unit/test.sh (hunk 1 above)
Comment (dustymabe) [reply tests/kola/ignition/systemd-enable-instance-unit/test.sh:25]: added

File: tests/kola/ignition/systemd-enable-instance-unit/test.sh [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+# make sure the unit ran and wrote 'foo' to the journal
+if [ $(journalctl -o cat -u echo@foo.service | sed -n 2p) != 'foo' ]; then
```
Comment (jlebon) [inline tests/kola/ignition/systemd-enable-instance-unit/test.sh:28]: We could also leverage the built-in `--grep`, e.g. `journalctl -u echo@foo.service --grep "I wrote foo"`. Obviously fine as is too!

**Review by jlebon — DISMISSED** [review]
LGTM

Should we restrict this to the qemu platform? Given that it seems highly unlikely to go wrong based on a specific platform and cloud testing is much more expensive time-wise.

**Comment by dustymabe** [comment]
> Should we restrict this to the qemu platform? Given that it seems highly unlikely to go wrong based on a specific platform and cloud testing is much more expensive time-wise.

done. I'll mark this to auto merge when tests pass

**Review by jlebon — APPROVED** [review]

**Pull request merged**